code/game/machinery/computer/security.dm

```diff
@@ -620,33 +620,35 @@ What a mess.*/
 
 						var/headerText = stripped_input(usr, "Please enter Poster Heading (Max 7 Chars):", "Print Wanted Poster", "WANTED", 8)
-
+						var/posternum = max(min((input("Number of posters to print (Max 5):","Number:",1) as num|null), 5), 1)
 						var/info = stripped_multiline_input(usr, "Please input a description for the poster:", "Print Wanted Poster", default_description, null)
 						if(info)
-							playsound(loc, 'sound/items/poster_being_created.ogg', 100, 1)
-							printing = 1
-							sleep(30)
-							if((istype(active1, /datum/data/record) && GLOB.data_core.general.Find(active1)))//make sure the record still exists.
-								var/obj/item/photo/photo = active1.fields["photo_front"]
-								new /obj/item/poster/wanted(loc, photo.picture.picture_image, wanted_name, info, headerText)
-							printing = 0
+							while(posternum--)
+								playsound(loc, 'sound/items/poster_being_created.ogg', 100, 1)
+								printing = 1
+								sleep(30)
+								if((istype(active1, /datum/data/record) && GLOB.data_core.general.Find(active1)))//make sure the record still exists.
+									var/obj/item/photo/photo = active1.fields["photo_front"]
+									new /obj/item/poster/wanted(loc, photo.picture.picture_image, wanted_name, info, headerText)
+								printing = 0
 			if("Print Missing")
 				if(!( printing ))
 					var/missing_name = stripped_input(usr, "Please enter an alias for the missing person:", "Print Missing Persons Poster", active1.fields["name"])
 					if(missing_name)
 						var/default_description = "A poster declaring [missing_name] to be a missing individual, missed by Nanotrasen. Report any sightings to security immediately."
 
 						var/headerText = stripped_input(usr, "Please enter Poster Heading (Max 7 Chars):", "Print Missing Persons Poster", "MISSING", 8)
-
+						var/posternum = max(min((input("Number of posters to print (Max 5):","Number:",1) as num|null), 5), 1)
 						var/info = stripped_multiline_input(usr, "Please input a description for the poster:", "Print Missing Persons Poster", default_description, null)
 						if(info)
-							playsound(loc, 'sound/items/poster_being_created.ogg', 100, 1)
-							printing = 1
-							sleep(30)
-							if((istype(active1, /datum/data/record) && GLOB.data_core.general.Find(active1)))//make sure the record still exists.
-								var/obj/item/photo/photo = active1.fields["photo_front"]
-								new /obj/item/poster/wanted/missing(loc, photo.picture.picture_image, missing_name, info, headerText)
-							printing = 0
+							while(posternum--)
+								playsound(loc, 'sound/items/poster_being_created.ogg', 100, 1)
+								printing = 1
+								sleep(30)
+								if((istype(active1, /datum/data/record) && GLOB.data_core.general.Find(active1)))//make sure the record still exists.
+									var/obj/item/photo/photo = active1.fields["photo_front"]
+									new /obj/item/poster/wanted/missing(loc, photo.picture.picture_image, missing_name, info, headerText)
+								printing = 0
```

Comment:
Put printing = TRUE before the loop and printing = FALSE after the loop (not inside of it)
Use a for loop instead of a while loop (for(var/i in 1 to posternum))